### PR TITLE
Store amounts using precise values in limit-orders

### LINKED ISSUE #1595: Limit price is lower than specified for Buy order
I placed a Buy Limit order with limit price =  13. 
When I [signed](https://protocol-explorer.dev.gnosisdev.com/gc/orders/0x518434c960c4435275555a885238525769961628113152001b70f9ced7ecd42f9fa3c00a92ec5f96b1ad2527ab41b3932efeda586389f682) an order, limit price became 12,999999999

**Expected**: price should be the same

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #1595

1. Store amounts using precise values instead of view values (view value = 5 USDC, unit value = 5000000 USDC)
2. Adjust decimals on switch token places
3. Adjust decimals on select currency

### CODE REVIEW

File: src/cow-react/modules/limitOrders/hooks/useApplyLimitRate.ts [hunk 1]
```diff
@@ -5,32 +5,39 @@ import { Field } from 'state/swap/actions'
...
+      if (!value || !Number(value) || !activeRate || activeRate.equalTo(0) || !inputCurrency || !outputCurrency) {
         return null
       }
 
-      let output: Fraction | null = null
-      const parsedValue = toFraction(value)
+      const { decimals: inputDecimals } = inputCurrency
+      const { decimals: outputDecimals } = outputCurrency
+
+      const parsedValue = adjustDecimals(
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/hooks/useApplyLimitRate.ts:25]: It's a fix for trading currencies with different decimals:
https://github.com/cowprotocol/cowswap/pull/1654/commits/0008155138b4988190635cc600cdd11edf980b91

**Comment by nenadV91** [comment]
For me, when you switch the currency, the price always flashes the old price after loader and then shows the new price.

**Review by nenadV91 — APPROVED** [review]
The issue with the decimals seems to be fixed
Code looks good but the mentioned issue should be fixed.

**Comment by shoom3301** [comment]
> For me, when you switch the currency, the price always flashes the old price after loader and then shows the new price.

@nenadV91 thanks! Can you show a video please? I can't reproduce

**Comment by elena-zh** [comment]
> > For me, when you switch the currency, the price always flashes the old price after loader and then shows the new price.
> 
> @nenadV91 thanks! Can you show a video please? I can't reproduce

I suppose, Nenad meant this: https://www.loom.com/share/fd213ecb18584ea9912a8fd70499d7fb

**Review by elena-zh — APPROVED** [review]
Besides the nitpick reported above, changes LGTM!
But I have added this issue into #1528 to address it later.

**Comment by nenadV91** [comment]
I meant what happens with the price in this case https://watch.screencastify.com/v/8SFLYiM3FaIvvH2XlSjG,

**Comment by shoom3301** [comment]
Hmmm, very interesting. I don't observer the behaviour. @elena-zh Do you?

**Comment by elena-zh** [comment]
> Hmmm, very interesting. I don't observer the behaviour. @elena-zh Do you?

I have recorded this issue and posted in above https://github.com/cowprotocol/cowswap/pull/1654#issuecomment-1342643426

**Review by anxolin — APPROVED** [review]
Approve, although i have to say i get a bit confused with so much hook logic

**Pull request merged**